# PCA of big vcf

First isolate chromosomes and trim them
```
module load StdEnv/2023  gcc/12.3 bcftools/1.19
bcftools filter -Oz --regions-file chr19.bed output.filtered.snps.AB.pass.removed.vep.vcf.gz -o output.filtered.snps.AB.pass.removed.vep_Chr19.vcf.gz
```

Now thin the file
```
#!/bin/sh
#SBATCH --job-name=vcftools
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=1:00:00
#SBATCH --mem=2gb
#SBATCH --output=vcftools.%J.out
#SBATCH --error=vcftools.%J.err
#SBATCH --account=def-ben

# sbatch 2024_vcftools_thin_vcf.sh pathandname_of_vcfgz_file number_of_bp_to_thin_by

module load StdEnv/2023 vcftools/0.1.16
vcftools --gzvcf ${1} --out ${1}_thin_5000.vcf --thin 5000 --recode
```

OK now do the pca and plot:
```R
# Note that a bug in the SNPrelate package means you need to quit R each time you reload
# a vcf file


#https://github.com/zhengxwen/SNPRelate/issues/13
#http://corearray.sourceforge.net/tutorials/SNPRelate/
#https://github.com/zhengxwen/SNPRelate/wiki/Preparing-Data

#GenotypeVCFs_noBSQR_filtered_aDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_xDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_aDNA_only_no_lowcoverage_individuals.vcf
setwd('/Users/Shared/Previously Relocated Items/Security/projects/2024_macaques/PCA')
library("devtools")
# BiocManager::install("gdsfmt")
library(gdsfmt)
# BiocManager::install("SNPRelate")
library(SNPRelate)



vcf.fn <- "output.filtered.snps.AB.pass.removed.vep.vcf.gz_chr1.bed.vcf.gz_thin_5000.vcf.recode.vcf"
snpgdsVCF2GDS(vcf.fn, "test.gds", method="biallelic.only",ignore.chr.prefix = c("chr","Scaffold"))

genofile = snpgdsOpen("test.gds", readonly=FALSE)
samp.annot<-data.frame(pop.group = c("MAU","MAU","MTH","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MTH","MTH","TON","MAU","TON","MAU","MAU","MTH","TON","MTH","MAU","MTH","MTH","TON","MTH","MTH","MTH","MTH","TON","MAU","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MAU","MTH","MTH","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","PAG","PAG","PAG","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","HEC","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","OCH","OCH","OCH","OCH","TON","OCH","OCH","OCH","MTH","MTH","MTH","MTH","MTH","MTH","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","TON","TON","TON","SIN","SIN","SIN","BOR","TON","NGA","BRU","HEC","HEC","HEC","MAU","MAU","MAU","MAU","MAU","BOR","SUM","BOR","BOR","NGA","NGA","NGE","TON","BOR","NGE","HEC","HEC","TON","TON","TON","TON","TON"))


add.gdsn(genofile, "sample.annot", val=samp.annot)

snpgdsSummary("test.gds")

#pca <- snpgdsPCA(genofile, num.thread=2)
#pc.percent <- pca$varprop*100
#head(round(pc.percent, 2))

# LD prunning
snpset <- snpgdsLDpruning(genofile, ld.threshold=1,  method = c("composite"),missing.rate=1, verbose = TRUE)
snpset.id <- unlist(snpset)
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))


tab <- data.frame(sample.id = pca$sample.id,
    EV1 = pca$eigenvect[,1],    # the first eigenvector
    EV2 = pca$eigenvect[,2],    # the second eigenvector
    EV3 = pca$eigenvect[,3],    # the second eigenvector
    stringsAsFactors = FALSE)
head(tab)

#plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
#text(tab$EV2, tab$EV1,labels=tab$sample.id, cex= 0.4)

library(ggplot2)
#ggplot(...)+...+ theme(axis.text.x = element_text(angle=60, hjust=1))
#devtools::install_github("slowkow/ggrepel")
library(ggrepel)

pdf("PCA_plot_chr1_E1E2.pdf",w=8, h=8, version="1.4", bg="transparent")
tab$Species <- c("MAU","MAU","MTH","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MTH","MTH","TON","MAU","TON","MAU","MAU","MTH","TON","MTH","MAU","MTH","MTH","TON","MTH","MTH","MTH","MTH","TON","MAU","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MAU","MTH","MTH","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","PAG","PAG","PAG","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","HEC","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","OCH","OCH","OCH","OCH","TON","OCH","OCH","OCH","MTH","MTH","MTH","MTH","MTH","MTH","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","TON","TON","TON","SIN","SIN","SIN","BOR","TON","NGA","BRU","HEC","HEC","HEC","MAU","MAU","MAU","MAU","MAU","BOR","SUM","BOR","BOR","NGA","NGA","NGE","TON","BOR","NGE","HEC","HEC","TON","TON","TON","TON","TON")
tab$samp.color <- c("blue","blue","purple","purple","blue","purple","purple","purple","purple","blue","blue","blue","purple","purple","purple","purple","purple","blue","purple","purple","red","blue","red","blue","blue","purple","red","purple","blue","purple","purple","red","purple","purple","purple","purple","red","blue","blue","purple","purple","purple","purple","blue","blue","purple","purple","purple","blue","blue","purple","purple","blue","purple","purple","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","green","green","green","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","red","red","orange","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","gray","gray","gray","gray","red","gray","gray","gray","purple","purple","purple","purple","purple","purple","yellow","yellow","yellow","yellow","yellow","yellow","red","red","red","red","red","black","black","black","brown","red","deeppink","darkorange","orange","orange","orange","blue","blue","blue","blue","blue","brown","yellow","brown","brown","deeppink","deeppink","deeppink","red","brown","deeppink","orange","orange","red","red","red","red","red")
tab$samp.fieldid <- c("WM029","WM073","WM094","PF012","WM002","WM059","WM078","WM097","PM013","WM003","WM018","WF098","WM004","WF033","WF080","WM100","PM016","WM005","WM062","WM081","WM126","WM006","WM127","WM008","WF022","WM064","WM129","PM020","WM023","WM065","WM084","WF131","PF026","WF050","WM067","WF085","WM133","WF012","WM026","WF051","WM068","WF087","PM001","WF013","WM027","WF052","WM069","WF088","PM002","WM015","WF053","WM070","PF004","PF021*","PM023","PM561","PM638","PM583","PM604","PM639","PM652","PF515","PM565","PM584","PM532","PM566","PF536","PM567","PF644","PM661","PM550","PM596","PM1006","pag_955","pag_1040","pag_820","6885","6890","6891","6892","6907","6916","6917","6966","6968","PM502","PM503","PM504","PF506","PM520","PM521","PM522","PM533","PF535","PF537","PF539","PF540","PM541","PM542","PF543","PM544","PM548","PM552","PF553","PM554","PM556","PM557","PF577","PF578","PM579","PF580","PF581","PF590","PM594","PM595","PF603","PF606","PM612","PM617","PM627","PF628","6997","6980","7016","7039","7057","7053","PM501","PF526","PM501","PM512","PM527","492","501","494","PM664","PM592","PF660","PF707","PF643","PF644","PF648","PF615","PF713","PM613","PM614","PM616","GumGum","Ngsang","PM665","Sukai","PF1001","PM1003","PM654","PF549","PM1206","PM1011","PF505","PF647","PF511","PF559","PF563","PF597","PF626")
d<-ggplot(data=tab, aes(x=EV1,y=EV2, label = samp.fieldid, color = samp.color)) +
    # label axis 
    labs(x=expression("Eigenvector 1"), y=expression("Eigenvector 2")) +
    # legend details
    scale_colour_manual(name="Population", values = c("purple"="purple",
                                                      "gray"="gray",
                                                      "green"="green",
                                                      "brown"="brown",
                                                      "blue"="blue",
                                                      "orange"="orange",
                                                      "black"="black",
                                                      "yellow"="yellow",
                                                      "deeppink"="deeppink",
                                                      "darkorange"="darkorange",
                                                      "red"="red"),
                        breaks=c("purple","gray","blue","orange","black","yellow","deeppink","darkorange","red"),
                        labels=c("MTH","OCH", "MAU", "HEC","SIN","SUM", "NGA","BRU","TON"))+
    # add points and fieldID labels
    geom_text_repel(aes(EV1,EV2, label=(samp.fieldid)), size = 3, max.overlaps = Inf) + geom_point(size=2) + 
    # change to cleaner theme
    #theme_classic(base_size = 8) +
    # make it clean
    theme_bw(base_size = 8)+ theme(panel.grid.minor=element_blank(),panel.grid.major=element_blank()) + 
    # italicize species names
    #theme(legend.text = element_text(face="italic"))+ 
    # move the legend
   # theme(legend.position = c(.8, .1)) +
    # add a title
    ggtitle("Principal Components Analsis") + 
    # remove boxes around legend symbols
    theme(legend.key = element_blank())
d
dev.off()









#https://github.com/zhengxwen/SNPRelate/issues/13
#http://corearray.sourceforge.net/tutorials/SNPRelate/
#https://github.com/zhengxwen/SNPRelate/wiki/Preparing-Data

#GenotypeVCFs_noBSQR_filtered_aDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_xDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_aDNA_only_no_lowcoverage_individuals.vcf
setwd('/Users/Shared/Previously\ Relocated\ Items/Security/projects/2021_Xborealis_sexchr_degen/PCA')
library("devtools")
library(gdsfmt)
library(SNPRelate)



#vcf.fn <- "XBwild_unfiltered_allchrs.vcf.gz_filtered.vcf.gz_filtered_removed_Chr7S_SL.recode.vcf"
#vcf.fn <- "XBwild_unfiltered_allchrs.vcf.gz_filtered.vcf.gz_filtered_removed_Chr7S_XMXVXB_notSL.recode.vcf"
vcf.fn <- "allsites_Chr7S_XMXVXB_SL.recode.vcf"
snpgdsVCF2GDS(vcf.fn, "test.gds", method="biallelic.only",ignore.chr.prefix = c("chr","Scaffold","chr9_"))

genofile = snpgdsOpen("test.gds", readonly=FALSE)
samp.annot<-data.frame(pop.group = c("V","V","V","V",
                                     "W","W","W","W","W","W",
                                     "W","W","W","W","W","W",
                                     "W","W","W","W","C","C",
                                     "C","C","C","C","C","C",
                                     "E","E","E","E","E","W",
                                     "W","W","W","W","W","W",
                                     "W","W","W","W","W","W",
                                     "W","W","C","C","C","C",
                                     "C","E","E","E","E","E","M"))


add.gdsn(genofile, "sample.annot", val=samp.annot)

snpgdsSummary("test.gds")

#pca <- snpgdsPCA(genofile, num.thread=2)
#pc.percent <- pca$varprop*100
#head(round(pc.percent, 2))

# LD pruning
snpset <- snpgdsLDpruning(genofile, ld.threshold=0.2,  method = c("composite"),missing.rate=0, verbose = TRUE)
snpset.id <- unlist(snpset)
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))




tab <- data.frame(sample.id = pca$sample.id,
                  EV1 = pca$eigenvect[,1],    # the first eigenvector
                  EV2 = pca$eigenvect[,2],    # the second eigenvector
                  EV3 = pca$eigenvect[,3],    # the second eigenvector
                  stringsAsFactors = FALSE)
head(tab)

#plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
#text(tab$EV2, tab$EV1,labels=tab$sample.id, cex= 0.4)

library(ggplot2)
#ggplot(...)+...+ theme(axis.text.x = element_text(angle=60, hjust=1))
#devtools::install_github("slowkow/ggrepel")
library(ggrepel)

pdf("PCA_plot_XB_W_E_Chr7S_lt23Mb_XMXVXB_E1E2.pdf",w=8, h=8, version="1.4", bg="transparent")
tab$Species <- c("V","V","V","V",
                 "W","W","W","W","W","W",
                 "W","W","W","W","W","W",
                 "W","W","W","W","C","C",
                 "C","C","C","C","C","C",
                 "E","E","E","E","E","W",
                 "W","W","W","W","W","W",
                 "W","W","W","W","W","W",
                 "W","W","C","C","C","C",
                 "C","E","E","E","E","E","M")
tab$samp.color <- c("purple","purple","purple","purple",
                    "blue","blue","blue","blue","blue","blue","blue","blue",
                    "blue","blue","blue","blue","blue","blue","blue","blue",
                    "gray","gray","gray","gray","gray","gray","gray","gray",
                    "red","red","red","red","red","blue","blue","blue","blue",
                    "blue","blue","blue","blue","blue","blue","blue","blue",
                    "blue","blue","blue","gray","gray","gray","gray","gray",
                    "red","red","red","red","red","green")
tab$samp.fieldid <- c("BJE4407_Xv","BJE4410_Xv","BJE4424_Xv","BJE4483_Xv","","",
                      "","",
                      "","",
                      "","",
                      "","",
                      "","",
                      "","",
                      "","",
                      "","","",
                      "","","",
                      "","","BJE4515 (F)",
                      "BJE4516 (F)","BJE4534 (F)",
                      "BJE4535 (F)","BJE4541 (F)",
                      "","",
                      "","",
                      "","",
                      "","","",
                      "","",
                      "","","",
                      "","","",
                      "","","",
                      "BJE4536 (M)","BJE4537 (M)",
                      "BJE4538 (M)","BJE4539 (M)",
                      "BJE4540 (M)","Xm")
#tab$samp.fieldid <- c("Fem_Cheplaskei_BJE4459","Fem_Cheplaskei_BJE4461",
#                      "Fem_Cheplaskei_BJE4470","Fem_Chesuwe_BJE4479",
#                      "Fem_Chesuwe_BJE4481","Fem_Eldoret_BJE4471",
#                      "Fem_Eldoret_BJE4472","Fem_Eldoret_BJE4474",
#                      "Fem_Eldoret_BJE4475","Fem_Eldoret_BJE4476",
#                      "Fem_Kiminini_BJE4429","Fem_Kiminini_BJE4433",
#                      "Fem_Lukhome_BJE4441","Fem_Lukhome_BJE4444",
#                      "Fem_Lukhome_BJE4445","Fem_Lukhome_BJE4446",
#                      "Fem_NMK_BJE4562","Fem_NMK_BJE4564","Fem_NMK_BJE4567",
#                      "Fem_NMK_BJE4568","Fem_Nakuru_BJE4363","Fem_Nakuru_BJE4364",
#                      "Fem_Nakuru_BJE4367","Fem_Nakuru_BJE4368","Fem_Wundanyi_BJE4515",
#                      "Fem_Wundanyi_BJE4516","Fem_Wundanyi_BJE4534",
#                      "Fem_Wundanyi_BJE4535","Fem_Wundanyi_BJE4541",
#                      "Mal_Cheplaskei_BJE4460","Mal_Cheplaskei_BJE4462",
#                      "Mal_Cheplaskei_BJE4465","Mal_Cheplaskei_BJE4469",
#                      "Mal_Chesuwe_BJE4477","Mal_Chesuwe_BJE4478",
#                      "Mal_Chesuwe_BJE4480","Mal_Eldoret_BJE4473","Mal_Kiminini_BJE4430",
#                      "Mal_Kiminini_BJE4431","Mal_Kiminini_BJE4432",
#                      "Mal_Kisumu_BJE4391","Mal_Lukhome_BJE4442","Mal_Lukhome_BJE4443",
#                      "Mal_Lukhome_BJE4447","Mal_NMK_BJE4563","Mal_Nakuru_BJE4365",
#                      "Mal_Nakuru_BJE4366","Mal_Nakuru_BJE4374","Mal_Nakuru_BJE4377",
#                      "Mal_Wundanyi_BJE4536","Mal_Wundanyi_BJE4537",
#                      "Mal_Wundanyi_BJE4538","Mal_Wundanyi_BJE4539",
#                      "Mal_Wundanyi_BJE4540")
# red is East
# gray is Central
# blue is West
d<-ggplot(data=tab, aes(x=EV1,y=EV2, label = samp.fieldid, color = samp.color)) +
    # label axis 
    labs(x=expression("Eigenvector 1"), y=expression("Eigenvector 2")) +
    # legend details
    scale_colour_manual(name="Species/Population", values = c("purple"="purple", 
                                                              "gray"="gray", "blue"="blue",
                                                              "red"="red", "green"="green"),
                        breaks=c("purple","gray","blue","red","green"),
                        labels=c("Xv","Central", "West", "East","Xm"))+
    # add points and fieldID labels
    geom_text_repel(aes(EV1,EV2, label=(samp.fieldid))) + geom_point(size=4) + 
    # change to cleaner theme
    theme_classic(base_size = 16) +
    # make it clean
    theme_bw()+ theme(panel.grid.minor=element_blank(),panel.grid.major=element_blank()) + 
    # italicize species names
    theme(legend.text = element_text(face="italic"))+ 
    # move the legend
    #theme(legend.position = c(.1, .1)) +
    # add a title
    ggtitle("Principal Components Analsis") + 
    # remove boxes around legend symbols
    theme(legend.key = element_blank())
d
dev.off()























#https://github.com/zhengxwen/SNPRelate/issues/13
#http://corearray.sourceforge.net/tutorials/SNPRelate/
#https://github.com/zhengxwen/SNPRelate/wiki/Preparing-Data

#GenotypeVCFs_noBSQR_filtered_aDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_xDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_aDNA_only_no_lowcoverage_individuals.vcf
setwd('/Users/Shared/Previously\ Relocated\ Items/Security/projects/2021_Xborealis_sexchr_degen/PCA')
library("devtools")
library(gdsfmt)
library(SNPRelate)



vcf.fn <- "mpileup_raw_wildBorealis_AustinGenome_Chrs7S_notSL_only.vcf.gz"
snpgdsVCF2GDS(vcf.fn, "test.gds", method="biallelic.only",ignore.chr.prefix = c("chr","Scaffold","chr9_"))

genofile = snpgdsOpen("test.gds", readonly=FALSE)
samp.annot<-data.frame(pop.group = c("W","W","W","W","W","V","W",
                                     "W","W","W","W","V","V","V",
                                     "V","V","V","W","W","V","V",
                                     "V","V","V","V","V","W","W",
                                     "W","W","W","W","W","W","W",
                                     "W","W","W","V","V","V","E",
                                     "E","E","E","E","W","W","W",
                                     "W","W","W","W","V","V","V",
                                     "V","W","V","V","V","V","V",
                                     "V","W","W","W","V","V","W",
                                     "V","V","V","V","V","W","W",
                                     "W","W","W","W","W","W","V",
                                     "V","V","E","E","E","E","E"))


add.gdsn(genofile, "sample.annot", val=samp.annot)

snpgdsSummary("test.gds")

#pca <- snpgdsPCA(genofile, num.thread=2)
#pc.percent <- pca$varprop*100
#head(round(pc.percent, 2))

# LD prinnung
snpset <- snpgdsLDpruning(genofile, ld.threshold=0.2,  method = c("composite"),missing.rate=0, verbose = TRUE)
snpset.id <- unlist(snpset)
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))




tab <- data.frame(sample.id = pca$sample.id,
                  EV1 = pca$eigenvect[,1],    # the first eigenvector
                  EV2 = pca$eigenvect[,2],    # the second eigenvector
                  EV3 = pca$eigenvect[,3],    # the second eigenvector
                  stringsAsFactors = FALSE)
head(tab)

#plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
#text(tab$EV2, tab$EV1,labels=tab$sample.id, cex= 0.4)

library(ggplot2)
#ggplot(...)+...+ theme(axis.text.x = element_text(angle=60, hjust=1))
#devtools::install_github("slowkow/ggrepel")
library(ggrepel)

pdf("PCA_plot_Xv_XB_W_E_Chr7S_gt23Mb.pdf",w=8, h=8, version="1.4", bg="transparent")
tab$Species <- c("W","W","W","W","W","V","W",
                 "W","W","W","W","V","V","V",
                 "V","V","V","W","W","V","V",
                 "V","V","V","V","V","W","W",
                 "W","W","W","W","W","W","W",
                 "W","W","W","V","V","V","E",
                 "E","E","E","E","W","W","W",
                 "W","W","W","W","V","V","V",
                 "V","W","V","V","V","V","V",
                 "V","W","W","W","V","V","W",
                 "V","V","V","V","V","W","W",
                 "W","W","W","W","W","W","V",
                 "V","V","E","E","E","E","E")
tab$samp.color <- c("blue","blue","blue","blue","blue","gray","blue","blue","blue","blue","blue","gray","gray","gray","gray","gray","gray","blue","blue","gray","gray","gray","gray","gray","gray","gray","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","blue","gray","gray","gray","red","red","red","red","red","blue","blue","blue","blue","blue","blue","blue","gray","gray","gray","gray","blue","gray","gray","gray","gray","gray","gray","blue","blue","blue","gray","gray","blue","gray","gray","gray","gray","gray","blue","blue","blue","blue","blue","blue","blue","blue","gray","gray","gray","red","red","red","red","red")
tab$samp.fieldid <- c("Fem_Cheplaskei_BJE4459_Xb","Fem_Cheplaskei_BJE4461_Xb","Fem_Cheplaskei_BJE4470_Xb","Fem_Chesuwe_BJE4479_Xb","Fem_Chesuwe_BJE4481_Xb","Fem_Chesuwe_BJE4486_Xv","Fem_Eldoret_BJE4471_Xb","Fem_Eldoret_BJE4472_Xb","Fem_Eldoret_BJE4474_Xb","Fem_Eldoret_BJE4475_Xb","Fem_Eldoret_BJE4476_Xb","Fem_Ikuywa_BJE4417_Xv","Fem_Ikuywa_BJE4420_Xv","Fem_Ikuywa_BJE4421_Xv","Fem_Kiminini_BJE4424_Xv","Fem_Kiminini_BJE4427_Xv","Fem_Kiminini_BJE4428_Xv","Fem_Kiminini_BJE4429_Xb","Fem_Kiminini_BJE4433_Xb","Fem_Kisumu_BJE4378_Xv","Fem_Kisumu_BJE4379_Xv","Fem_Kisumu_BJE4389_Xv","Fem_Longisa_BJE4450_Xv","Fem_Longisa_BJE4453_Xv","Fem_Longisa_BJE4455_Xv","Fem_Longisa_BJE4457_Xv","Fem_Lukhome_BJE4441_Xb","Fem_Lukhome_BJE4444_Xb","Fem_Lukhome_BJE4445_Xb","Fem_Lukhome_BJE4446_Xb","Fem_Nakuru_BJE4363_Xb","Fem_Nakuru_BJE4364_Xb","Fem_Nakuru_BJE4367_Xb","Fem_Nakuru_BJE4368_Xb","Fem_NMK_BJE4562_Xb","Fem_NMK_BJE4564_Xb","Fem_NMK_BJE4567_Xb","Fem_NMK_BJE4568_Xb","Fem_Sichirai_BJE4397_Xv","Fem_Sichirai_BJE4398_Xv","Fem_Sichirai_BJE4407_Xv","Fem_Wundanyi_BJE4515_Xb","Fem_Wundanyi_BJE4516_Xb","Fem_Wundanyi_BJE4534_Xb","Fem_Wundanyi_BJE4535_Xb","Fem_Wundanyi_BJE4541_Xb","Mal_Cheplaskei_BJE4460_Xb","Mal_Cheplaskei_BJE4462_Xb","Mal_Cheplaskei_BJE4465_Xb","Mal_Cheplaskei_BJE4469_Xb","Mal_Chesuwe_BJE4477_Xb","Mal_Chesuwe_BJE4478_Xb","Mal_Chesuwe_BJE4480_Xb","Mal_Chesuwe_BJE4482_Xv","Mal_Chesuwe_BJE4483_Xv","Mal_Chesuwe_BJE4484_Xv","Mal_Chesuwe_BJE4485_Xv","Mal_Eldoret_BJE4473_Xb","Mal_Ikuywa_BJE4410_Xv","Mal_Ikuywa_BJE4411_Xv","Mal_Ikuywa_BJE4414_Xv","Mal_Ikuywa_BJE4418_Xv","Mal_Kiminini_BJE4425_Xv","Mal_Kiminini_BJE4426_Xv","Mal_Kiminini_BJE4430_Xb","Mal_Kiminini_BJE4431_Xb","Mal_Kiminini_BJE4432_Xb","Mal_Kisumu_BJE4384_Xv","Mal_Kisumu_BJE4390_Xv","Mal_Kisumu_BJE4391_Xb","Mal_Kisumu_BJE4392_Xv","Mal_Kisumu_BJE4497_Xv","Mal_Longisa_BJE4449_Xv","Mal_Longisa_BJE4451_Xv","Mal_Longisa_BJE4452_Xv","Mal_Lukhome_BJE4442_Xb","Mal_Lukhome_BJE4443_Xb","Mal_Lukhome_BJE4447_Xb","Mal_Nakuru_BJE4365_Xb","Mal_Nakuru_BJE4366_Xb","Mal_Nakuru_BJE4374_Xb","Mal_Nakuru_BJE4377_Xb","Mal_NMK_BJE4563_Xb","Mal_Sichirai_BJE4400_Xv","Mal_Sichirai_BJE4401_Xv","Mal_Sichirai_BJE4402_Xv","Mal_Wundanyi_BJE4536_Xb","Mal_Wundanyi_BJE4537_Xb","Mal_Wundanyi_BJE4538_Xb","Mal_Wundanyi_BJE4539_Xb","Mal_Wundanyi_BJE4540_Xb")
# red is East
# gray is vict
# blue is West
d<-ggplot(data=tab, aes(x=EV1,y=-EV2, label = samp.fieldid, color = samp.color)) +
    # label axis 
    labs(x=expression("Eigenvector 1"), y=expression("Eigenvector 2")) +
    # legend details
    scale_colour_manual(name="Species/Population", values = c("gray"="gray", "blue"="blue","red"="red"),breaks=c("gray","blue","red"),labels=c("X. victorianus", "X. borealis West", "X. borealis East"))+
    # add points and fieldID labels
    geom_text_repel(aes(EV1,-EV2, label=(samp.fieldid))) + geom_point(size=4) + 
    # change to cleaner theme
    theme_classic(base_size = 16) +
    # make it clean
    theme_bw()+ theme(panel.grid.minor=element_blank(),panel.grid.major=element_blank()) + 
    # italicize species names
    theme(legend.text = element_text(face="italic"))+ 
    # move the legend
    theme(legend.position = c(.1, .1)) +
    # add a title
    ggtitle("Principal Components Analsis") + 
    # remove boxes around legend symbols
    theme(legend.key = element_blank())
d
dev.off()

```
