# PCA of big vcf

First isolate chromosomes and trim them
```
module load StdEnv/2023  gcc/12.3 bcftools/1.19
bcftools filter -Oz --regions-file chr19.bed output.filtered.snps.AB.pass.removed.vep.vcf.gz -o output.filtered.snps.AB.pass.removed.vep_Chr19.vcf.gz
```

Now thin the file
```
#!/bin/sh
#SBATCH --job-name=vcftools
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=1:00:00
#SBATCH --mem=2gb
#SBATCH --output=vcftools.%J.out
#SBATCH --error=vcftools.%J.err
#SBATCH --account=def-ben

# sbatch 2024_vcftools_thin_vcf.sh pathandname_of_vcfgz_file number_of_bp_to_thin_by

module load StdEnv/2023 vcftools/0.1.16
vcftools --gzvcf ${1} --out ${1}_thin_5000.vcf --thin 5000 --recode
```

OK now do the pca and plot:
```R
# Note that a bug in the SNPrelate package means you need to quit R each time you reload
# a vcf file


#https://github.com/zhengxwen/SNPRelate/issues/13
#http://corearray.sourceforge.net/tutorials/SNPRelate/
#https://github.com/zhengxwen/SNPRelate/wiki/Preparing-Data

#GenotypeVCFs_noBSQR_filtered_aDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_xDNA_only.vcf.gz
#GenotypeVCFs_noBSQR_filtered_aDNA_only_no_lowcoverage_individuals.vcf
setwd('/Users/Shared/Previously Relocated Items/Security/projects/2024_macaques/PCA')
library("devtools")
# BiocManager::install("gdsfmt")
library(gdsfmt)
# BiocManager::install("SNPRelate")
library(SNPRelate)



vcf.fn <- "output.filtered.snps.AB.pass.removed.vep.vcf.gz_chr1.bed.vcf.gz_thin_5000.vcf.recode.vcf"
snpgdsVCF2GDS(vcf.fn, "test.gds", method="biallelic.only",ignore.chr.prefix = c("chr","Scaffold"))

genofile = snpgdsOpen("test.gds", readonly=FALSE)
samp.annot<-data.frame(pop.group = c("MAU","MAU","MTH","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MTH","MTH","TON","MAU","TON","MAU","MAU","MTH","TON","MTH","MAU","MTH","MTH","TON","MTH","MTH","MTH","MTH","TON","MAU","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MAU","MTH","MTH","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","PAG","PAG","PAG","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","HEC","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","OCH","OCH","OCH","OCH","TON","OCH","OCH","OCH","MTH","MTH","MTH","MTH","MTH","MTH","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","TON","TON","TON","SIN","SIN","SIN","BOR","TON","NGA","BRU","HEC","HEC","HEC","MAU","MAU","MAU","MAU","MAU","BOR","SUM","BOR","BOR","NGA","NGA","NGE","TON","BOR","NGE","HEC","HEC","TON","TON","TON","TON","TON"))


add.gdsn(genofile, "sample.annot", val=samp.annot)

snpgdsSummary("test.gds")

#pca <- snpgdsPCA(genofile, num.thread=2)
#pc.percent <- pca$varprop*100
#head(round(pc.percent, 2))

# LD prunning
snpset <- snpgdsLDpruning(genofile, ld.threshold=1,  method = c("composite"),missing.rate=1, verbose = TRUE)
snpset.id <- unlist(snpset)
pca <- snpgdsPCA(genofile, snp.id=snpset.id, num.thread=2)
pc.percent <- pca$varprop*100
head(round(pc.percent, 2))


tab <- data.frame(sample.id = pca$sample.id,
    EV1 = pca$eigenvect[,1],    # the first eigenvector
    EV2 = pca$eigenvect[,2],    # the second eigenvector
    EV3 = pca$eigenvect[,3],    # the second eigenvector
    stringsAsFactors = FALSE)
head(tab)

#plot(tab$EV2, tab$EV1, xlab="eigenvector 2", ylab="eigenvector 1")
#text(tab$EV2, tab$EV1,labels=tab$sample.id, cex= 0.4)

library(ggplot2)
#ggplot(...)+...+ theme(axis.text.x = element_text(angle=60, hjust=1))
#devtools::install_github("slowkow/ggrepel")
library(ggrepel)

pdf("PCA_plot_chr1_E1E2.pdf",w=8, h=8, version="1.4", bg="transparent")
tab$Species <- c("MAU","MAU","MTH","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MAU","MTH","MTH","MTH","MTH","MAU","MTH","MTH","TON","MAU","TON","MAU","MAU","MTH","TON","MTH","MAU","MTH","MTH","TON","MTH","MTH","MTH","MTH","TON","MAU","MAU","MTH","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MTH","MAU","MAU","MTH","MTH","MAU","MTH","MTH","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","PAG","PAG","PAG","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","HEC","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","TON","OCH","OCH","OCH","OCH","TON","OCH","OCH","OCH","MTH","MTH","MTH","MTH","MTH","MTH","SUM","SUM","SUM","SUM","SUM","SUM","TON","TON","TON","TON","TON","SIN","SIN","SIN","BOR","TON","NGA","BRU","HEC","HEC","HEC","MAU","MAU","MAU","MAU","MAU","BOR","SUM","BOR","BOR","NGA","NGA","NGE","TON","BOR","NGE","HEC","HEC","TON","TON","TON","TON","TON")
tab$samp.color <- c("blue","blue","purple","purple","blue","purple","purple","purple","purple","blue","blue","blue","purple","purple","purple","purple","purple","blue","purple","purple","red","blue","red","blue","blue","purple","red","purple","blue","purple","purple","red","purple","purple","purple","purple","red","blue","blue","purple","purple","purple","purple","blue","blue","purple","purple","purple","blue","blue","purple","purple","blue","purple","purple","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","green","green","green","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","yellow","red","red","orange","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","red","gray","gray","gray","gray","red","gray","gray","gray","purple","purple","purple","purple","purple","purple","yellow","yellow","yellow","yellow","yellow","yellow","red","red","red","red","red","black","black","black","brown","red","deeppink","darkorange","orange","orange","orange","blue","blue","blue","blue","blue","brown","yellow","brown","brown","deeppink","deeppink","deeppink","red","brown","deeppink","orange","orange","red","red","red","red","red")
tab$samp.fieldid <- c("WM029","WM073","WM094","PF012","WM002","WM059","WM078","WM097","PM013","WM003","WM018","WF098","WM004","WF033","WF080","WM100","PM016","WM005","WM062","WM081","WM126","WM006","WM127","WM008","WF022","WM064","WM129","PM020","WM023","WM065","WM084","WF131","PF026","WF050","WM067","WF085","WM133","WF012","WM026","WF051","WM068","WF087","PM001","WF013","WM027","WF052","WM069","WF088","PM002","WM015","WF053","WM070","PF004","PF021*","PM023","PM561","PM638","PM583","PM604","PM639","PM652","PF515","PM565","PM584","PM532","PM566","PF536","PM567","PF644","PM661","PM550","PM596","PM1006","pag_955","pag_1040","pag_820","6885","6890","6891","6892","6907","6916","6917","6966","6968","PM502","PM503","PM504","PF506","PM520","PM521","PM522","PM533","PF535","PF537","PF539","PF540","PM541","PM542","PF543","PM544","PM548","PM552","PF553","PM554","PM556","PM557","PF577","PF578","PM579","PF580","PF581","PF590","PM594","PM595","PF603","PF606","PM612","PM617","PM627","PF628","6997","6980","7016","7039","7057","7053","PM501","PF526","PM501","PM512","PM527","492","501","494","PM664","PM592","PF660","PF707","PF643","PF644","PF648","PF615","PF713","PM613","PM614","PM616","GumGum","Ngsang","PM665","Sukai","PF1001","PM1003","PM654","PF549","PM1206","PM1011","PF505","PF647","PF511","PF559","PF563","PF597","PF626")
d<-ggplot(data=tab, aes(x=EV1,y=EV2, label = samp.fieldid, color = samp.color)) +
    # label axis 
    labs(x=expression("Eigenvector 1"), y=expression("Eigenvector 2")) +
    # legend details
    scale_colour_manual(name="Population", values = c("purple"="purple",
                                                      "gray"="gray",
                                                      "green"="green",
                                                      "brown"="brown",
                                                      "blue"="blue",
                                                      "orange"="orange",
                                                      "black"="black",
                                                      "yellow"="yellow",
                                                      "deeppink"="deeppink",
                                                      "darkorange"="darkorange",
                                                      "red"="red"),
                        breaks=c("purple","gray","blue","orange","black","yellow","deeppink","darkorange","red"),
                        labels=c("MTH","OCH", "MAU", "HEC","SIN","SUM", "NGA","BRU","TON"))+
    # add points and fieldID labels
    geom_text_repel(aes(EV1,EV2, label=(samp.fieldid)), size = 3, max.overlaps = Inf) + geom_point(size=2) + 
    # change to cleaner theme
    #theme_classic(base_size = 8) +
    # make it clean
    theme_bw(base_size = 8)+ theme(panel.grid.minor=element_blank(),panel.grid.major=element_blank()) + 
    # italicize species names
    #theme(legend.text = element_text(face="italic"))+ 
    # move the legend
   # theme(legend.position = c(.8, .1)) +
    # add a title
    ggtitle("Principal Components Analsis") + 
    # remove boxes around legend symbols
    theme(legend.key = element_blank())
d
dev.off()
```
